projects/adventure/adv.py

```diff
@@ -22,7 +22,56 @@
 player = Player("Name", world.startingRoom)
 
 # Fill this out
+
+'''
+- record traversed rooms
+- add current room
+- 
+- make sure on way back to room you dont have to go retrace steps to go back into each room to get to starter
+as that would add a lot of unneccesary moves (brady)
+'''
+
+
+def oppositeDirection(direction):
+    if direction is 'n':
+        return 's' 
+    elif direction is 's':
+        return 'n'
+    elif direction is 'e':
+        return 'w'
+    elif direction is 'w':
+        return 'e'
+
 traversalPath = []
+entered = {}
+entered[player.currentRoom.id] = player.currentRoom.getExits()
+pathBack = []
+
+
+while len(entered) < (len(roomGraph) - 1):
+    # record room entered if not entered 
```

Comment:
Consider how we might structure this loop if we did not know/have access to the number of rooms in the graph.